.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,66 @@
+# This code is a Qiskit project.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+name: PR preview
+on:
+  pull_request:
```

Comment:
In https://github.com/Qiskit/documentation/pull/1595, we specified which PR events to run this on. That's because we cared about doing clean up very eagerly, as soon as the PR was closed. With this new GH Pages approach, we only do cleanup on pushes to `main`, that it will see if there are any stale folders in `gh-pages` branch. It's not a very big deal to have an extra folder hanging around for a bit.